### ISSUE TITLE
Future of Coding Weekly 2021/06 Week 1

### ISSUE THREAD

**Issue by marianoguerra** [marianoguerra/future-of-coding-weekly#79]
- [x] Change Month and Week Number
- [x] focWeekExport 2021-04-19 2021-04-26
- [x] Summary
- [ ] Hashtags
- [x] Check that comment links work (push weekly dump with channel summaries)
- [x] Check to mention right person for moved messages
- [x] Update Search Index
- [x] Download New Attachments

- https://tinyletter.com/
- https://tinyletter.com/marianoguerra/letters/
- http://localhost:8000/history/
- https://marianoguerra.github.io/future-of-coding-weekly/
- https://stackedit.io/

**Comment by marianoguerra** [comment]
🐢 Referential Transparency All The Way Down 🎙FoC Podcast 🛸 Logic Relational Programming 💡 Inevitability of Objects

# Two Minute Week

🎥 [Referential transparency atop an imperative substrate](https://archive.org/details/akkartik-mu-2021-05-31) via [Kartik Agaram](https://github.com/akkartik/mu)

[🧵 conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2021/06/W1/two-minute-week.html#2021-06-01T05:08:14.014Z)


![mu-sandbox.png](http://history.futureofcoding.org/history/msg_files/F02/F024A1U286L.png)

**Comment by marianoguerra** [comment]
# Our Work

🔗 [Create a new App - Gadget](https://www.loom.com/share/732816bb327c4155b61199d2d1733c56) via **Mo**

[🧵 conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2021/06/W1/share-your-work.html#2021-06-01T12:38:46.030Z)

Hey folks, after a long year of building and rebuilding [Gadget](https://gadget.dev/), we're finally gearing up for a launch. We're looking for early alpha users if anyone is interested ([drop your email on the site](https://gadget.dev/) or DM me).



Problem: Web development is slower and more expensive than it should be, because of the endless, repetitive boilerplate/glue code that developers have to manouver.

Challenge: Build a platform that gives developers a set of higher level building blocks, while providing them with the ability to drop down to a lower level when the abstractions don't work.

Solution: Gadget, a preconfigured, bundled development platform (IDE +++) that gives you a set of components with sane defaults that you can override, in code,  designed to meet the needs of modern web application developers.



Basically, Gadget is our vision of what "low code for devs" should actually feel like (for web app development). An experience where the computer is responsible for all the annoying busywork the developer doesn't want to do,  while leaving the engineer to make all the engineering decisions.



Here's a not-so-short video (20 min) where I build, deploy and test a declarative API for the backend of a Kanban board: [Create a new App - Gadget](https://www.loom.com/share/732816bb327c4155b61199d2d1733c56)



Want to try it out? DM me or drop your email here: [https://gadget.dev/](https://gadget.dev/)

**Comment by marianoguerra** [comment]
📝 [Dion Roadmap](http://dion.systems/blog_0002_roadmap.html) via **Allen Webster**

[🧵 conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2021/06/W1/share-your-work.html#2021-06-01T15:31:12.032Z)

We (the Dion team) just published our website and a blog post about our approach to experimenting and iterating on a source code format

**Comment by marianoguerra** [comment]
🎙🎂 [Future of Coding Podcast Episode 50: Mary Rose Cook](https://futureofcoding.org/episodes/050) via [Ivan Reese](https://twitter.com/spiralganglion)

[🧵 conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2021/06/W1/share-your-work.html#2021-06-04T01:26:34.043Z)

Today, the guest is [Mary Rose Cook](https://maryrosecook.com), among whose thousand-and-one side projects you'll find [Isla](http://islalanguage.org) and [Code Lauren](http://codelauren.com). Both are experimental programming environments with their own bespoke languages, focussed on learnability and visualization. In the case of Code Lauren, we also see another theme of Mary's work — taking the largely invisible processing done by compilers / interpreters and making it visible and interactive. Mary also made a totally awful game that I hate so much, but to find out what that is you'll just have to listen to the show (or read the largely unedited transcript — again, sorry Jack).

💬 [Konrad Hinsen](https://khinsen.net/)

[🧵 conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2021/06/W1/thinking-together.html#2021-06-05T08:39:32.038Z)

This is a great episode, with touches on many topics. One of them was so much in collision with my own ideas for the future of coding that I am tempted to launch a discussion on it. It's the notion of an "escape hatch" from a programming environment.



Overall, there's too much of a "let's build a universe" attitude in computing, and the idea of a programming environment ideally being complete and mostly closed is a reflection of it. I don't want  an escape hatch for going elsewhere if I absolutely have to. I want bridges to other programming environments, for my own use and for connecting with other people's work.



My idea of the future of coding is more like living in physical space. I live in a house, which has multiple rooms. In some rooms, I do things alone, in others with family members. The house is part of a neighborhood, where I do things with neighbours. The neighborhood is part of a town, and so on. Each compartment provides just the right environment for some activity, but none locks me in, wishing for an escape hatch.

**Comment by marianoguerra** [comment]
# Thinking Together

💬 [Mariano Guerra](https://twitter.com/warianoguerra)

[🧵 conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2021/06/W1/thinking-together.html#2021-06-06T11:48:42.044Z)

How much intersection/overlap do you think there is between Future of Coding and Tools for Thought "movements"?

**Comment by marianoguerra** [comment]
💬 **Rob Haisfield**

[🧵 conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2021/06/W1/thinking-together.html#2021-06-06T13:38:42.046Z)

What do you gain from being able to run a program in reverse? Just looking at [Ivan Reese](https://twitter.com/spiralganglion)’s Hest but curious more generally

**Comment by marianoguerra** [comment]
# Content

🎥 [Nushell 0.32](https://www.youtube.com/watch?v=AqYxhJKblvY) via [Mariano Guerra](https://twitter.com/warianoguerra)

[🧵 conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2021/06/W1/linking-together.html#2021-06-01T21:13:59.034Z)

![Thumbnail](https://img.youtube.com/vi/AqYxhJKblvY/hqdefault.jpg)

**Comment by marianoguerra** [comment]
🎥 [William Byrd on "The Most Beautiful Program Ever Written"](https://www.youtube.com/watch?v=OyfBQmvr2Hc) via [Kartik Agaram](https://github.com/akkartik/mu)

[🧵 conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2021/06/W1/linking-together.html#2021-06-02T05:21:31.035Z)

![Thumbnail](https://img.youtube.com/vi/OyfBQmvr2Hc/hqdefault.jpg)

William Byrd time: [William Byrd on Logic and Relational Programming, miniKanren](https://www.infoq.com/interviews/byrd-relational-programming-minikanren)


After being vaguely aware of miniKanren for a decade or so, I seem to finally be colliding with it when I'm ready for it (thanks to [this thread](https://futureofcoding.slack.com/archives/C5T9GPWFL/p1622304307031200?thread_ts=1622048775.021900&cid=C5T9GPWFL) and [this HN conversation](https://news.ycombinator.com/item?id=27359963))

**Comment by marianoguerra** [comment]
🐦 [Tweet from @jjcarett2](https://twitter.com/jjcarett2/status/1399473063515889672) via [Andreas S](https://twitter.com/curious_reader)

[🧵 conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2021/06/W1/linking-together.html#2021-06-03T09:23:19.039Z)

Has this been discussed here before? Thoughts? 

> 🐦 [Jacques Carette](https://twitter.com/@jjcarett2): I recently learned of [@JAldrichPL](https://twitter.com/JAldrichPL) 's essay ["The Power of Interoperability: Why Objects Are Inevitable"](http://www.cs.cmu.edu/~aldrich/papers/objects-essay.pdf). Saved it for later reading [now done]. I was all revved up to hate it, and maybe even rant. My rant-readers will be disappointed: I rather liked it.

**Comment by marianoguerra** [comment]
https://tinyletter.com/marianoguerra/letters/future-of-coding-weekly-2021-06-week-1
